app/main/views/agreements.py

```diff
@@ -0,0 +1,78 @@
+from flask import jsonify, abort
+from .. import main
+from sqlalchemy.exc import IntegrityError
+from dmapiclient.audit import AuditTypes
+from ...models import (
+    db, FrameworkAgreement, AuditEvent
+)
+
+from ...utils import (
+    get_json_from_request,
+    json_has_required_keys,
+    json_has_keys,
+    validate_and_return_updater_request,
+)
+
+from ...supplier_utils import validate_agreement_details_data
+
+
+@main.route('/agreements/<int:agreement_id>', methods=['GET'])
+def get_framework_agreement(agreement_id):
+    framework_agreement = FrameworkAgreement.query.filter(FrameworkAgreement.id == agreement_id).first_or_404()
+
+    return jsonify(agreement=framework_agreement.serialize())
+
+
+@main.route('/agreements/<int:agreement_id>', methods=['POST'])
+def update_framework_agreement(agreement_id):
+    framework_agreement = FrameworkAgreement.query.filter(FrameworkAgreement.id == agreement_id).first_or_404()
+    framework_agreement_details = framework_agreement.supplier_framework.framework.framework_agreement_details
+
+    json_payload = get_json_from_request()
+    updater_json = validate_and_return_updater_request()
+    json_has_required_keys(json_payload, ["agreement"])
+    update_json = json_payload["agreement"]
+
+    json_has_keys(update_json, optional_keys=['signedAgreementDetails', 'signedAgreementPath'])
+
+    if (
+        framework_agreement.signed_agreement_returned_at
+        and ('signedAgreementDetails' in update_json or 'signedAgreementPath' in update_json)
+    ):
+        abort(400, "Can not update signedAgreementDetails or signedAgreementPath if agreement has been signed")
```

Comment:
@allait Am I right in thinking that this check should be removed? If we wish to do the smallest refactoring step possible and move all existing behaviour into our new model then this behaviour should be retained (because we currently allow users to restart the process in our frontend)